src/main/java/me/coley/recaf/util/IOUtil.java

```diff
@@ -395,4 +458,36 @@ public static Path toPath(File file) {
     public static String toString(Path path) {
         return path.toAbsolutePath().normalize().toString();
     }
+
+    /**
+     * @param bytes byte array to check.
+     * @return {@code true} if byte array represents
+     * a zip file header.
+     */
+    public static boolean isZipHeader(byte[] bytes) {
+        return bytes[0] == (byte) 0x50 && bytes[1] == (byte) 0x4B
+                && bytes[2] == (byte) 0x03 && bytes[3] == (byte) 0x04;
+    }
+
+    /**
+     * @param bytes byte array to check.
+     * @return {@code true} if byte array represents
+     * a class file header.
+     */
+    public static boolean isClassHeader(byte[] bytes) {
+        return bytes[0] == (byte) 0xCA && bytes[1] == (byte) 0xFE
+                && bytes[2] == (byte) 0xBA && bytes[3] == (byte) 0xBE;
+    }
+
+    /**
+     * @param bytes byte array to check.
+     * @return {@code true} if byte array represents
+     * jar file signature.
+     */
+    public static boolean isJarSignature(byte[] bytes) {
+        if (bytes.length < 2) {
+            return false;
+        }
+        return bytes[0] == -2 && bytes[1] == -54;
```

Comment:
The other methods don't have length checks for the input byte array, do we want to add this check to the other methods to prevent possible `IndexOutOfBoundsException`?